modules/core/shared/src/main/scala/gem/config/DynamicConfig.scala

```diff
@@ -188,17 +207,33 @@ object DynamicConfig {
   }
 
   final case class Gnirs(
-    acquisitionMirror:    GnirsAcquisitionMirror,
-    camera:               GnirsCamera,
-    decker:               GnirsDecker,
-    disperser:            GnirsDisperser,
-    exposureTime:         Duration,
-    filter:               GnirsFilter,
-    fpu:                  Either[GnirsFpuOther, GnirsFpuSlit],
-    prism:                GnirsPrism,
-    readMode:             GnirsReadMode,
-    wavelength:           Wavelength
-  ) extends DynamicConfig.Impl(Instrument.Gnirs)
+    acquisitionMirror: GnirsAcquisitionMirror,
+    camera:            GnirsCamera,
+    decker:            GnirsDecker,
+    disperser:         GnirsDisperser,
+    exposureTime:      Duration,
+    filter:            GnirsFilter,
+    fpu:               Either[GnirsFpuOther, GnirsFpuSlit],
+    prism:             GnirsPrism,
+    readMode:          GnirsReadMode,
+    wavelength:        Wavelength
+  ) extends DynamicConfig.Impl(Instrument.Gnirs) {
+
+    /** Returns the smart gcal search key for this GNIRS configuration. */
+    def key(s: StaticConfig.Gnirs): SmartGcalKey.GnirsSearch =
+      SmartGcalKey.GnirsSearch(
+        SmartGcalKey.Gnirs(
+          acquisitionMirror,
+          camera.pixelScale,
+          disperser,
+          fpu,
+          prism,
+          s.wellDepth
+        ),
+        Some(wavelength)
```

Comment:
Perhaps the wavelength parameter doesn't need to be an `Option` if it is always `Some`?